pkg/client/fake/node_client.go

```diff
@@ -0,0 +1,52 @@
+package fake
+
+import (
+	"context"
+
+	v1 "k8s.io/api/core/v1"
+
+	"sigs.k8s.io/container-object-storage-interface-api/apis/objectstorage.k8s.io/v1alpha1"
+	"sigs.k8s.io/container-object-storage-interface-csi-adapter/pkg/client"
+)
+
+var _ client.NodeClient = &FakeNodeClient{}
+
+type FakeNodeClient struct {
+	MockGetBAR func(ctx context.Context, barName, barNs string) (*v1alpha1.BucketAccessRequest, error)
+	MockGetBA  func(ctx context.Context, baName string) (*v1alpha1.BucketAccess, error)
+	MockGetBR  func(ctx context.Context, brName, brNs string) (*v1alpha1.BucketRequest, error)
+	MockGetB   func(ctx context.Context, bName string) (*v1alpha1.Bucket, error)
+
+	MockGetResources func(ctx context.Context, barName, barNs string) (bkt *v1alpha1.Bucket, ba *v1alpha1.BucketAccess, secret *v1.Secret, err error)
+
+	MockAddBAFinalizer    func(ctx context.Context, ba *v1alpha1.BucketAccess, BAFinalizer string) error
+	MockRemoveBAFinalizer func(ctx context.Context, ba *v1alpha1.BucketAccess, BAFinalizer string) error
+}
+
+func (f FakeNodeClient) GetBAR(ctx context.Context, barName, barNs string) (*v1alpha1.BucketAccessRequest, error) {
```

Comment:
These look like helper method, rather than core methods of an abstraction. Why define an interface for it? Why not just pass in a mock bucket client ?

Reply:
Do you mean that one of the parameters should be the bucket client, and then the method could handle the rest? That would work as well, although, in terms of a design, it seemed better to be to isolate the methods to a specific struct that wraps the underlying client. This way we could also include the other Kubernetes.Clientset as well, and it would be opaque to the consumer (in this case, either the tests or the the actual NodeServer)